src/lib/core/components/visualizations/implementations/BarplotVisualization.tsx

```diff
@@ -124,6 +126,13 @@ function BarplotViz(props: VisualizationProps) {
     [updateConfiguration, vizConfig]
   );
 
+  // update currentPlotfilters with the latest filters at fullscreen mode
+  useEffect(() => {
+    updateVizConfig({
+      currentPlotFilters: filters,
+    });
+  }, [filters]);
```

Comment:
Have you tried putting this effect into the root `FullscreenVisualization` component? If doing that works, that would save us the hassle of needing to include this effect in every new visualization.

(Sidenote: we should consider a similar DRY-up of the `updateThumbnail` effects. That will be trickier because currently, each viz's `updateThumbnail` effect has different dependencies.)

Reply:
@jtlong3rd Actually I was debating where to put the new props, `currentPlotFilters`, either configuration or descriptor as you said. But I like your idea to place the props at descriptor as it does not need to change all plot vizs. I have changed to use the useEffect at `FullscreenVisualization` component and it seems to work fine as well 🎉